docs/codeql/codeql-cli/analyzing-databases-with-the-codeql-cli.rst

```diff
@@ -142,6 +143,13 @@ These are stored alongside the code scanning suites with names of the form: ``<l
 For information about creating custom query suites, see ":doc:`Creating
 CodeQL query suites <creating-codeql-query-suites>`."
 
+Diagnostic information
+......................
+
+The code scanning query suites include additional diagnostic queries. When the database analysis is complete, the CLI generates the results file and reports any diagnostic data to standard output. If you choose to generate SARIF output, the diagnostic data is also included as `notification objects <https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317894>`__ in the SARIF file.
```

Comment:
There are two kinds of diagnostic data we introduced:
- summary metrics are stored in the properties bag on each `run` object, in a new property called `metricResults`. (This is our own addition to the SARIF spec. I don't think we have to document in detail exactly what the format is yet, in case we need to adjust it.)
- extractor diagnostics are `notification` objects in `toolExecutionNotifications`, as you describe.

Reply:
Hmm. Thank you for explaining. The information here was taken from the [changelog](https://github.com/github/semmle-code/blob/main/odasa-buildutils/codeql-resources/CHANGELOG.md#features-added).

Based on your comment above, I've generalized this section and removed the information about notification objects for now.